opentracing-api/src/main/java/io/opentracing/SpanScheduler.java

```diff
@@ -0,0 +1,87 @@
+package io.opentracing;
+
+/**
+ * Scheduler allows an existing (possibly thread-local-aware) execution context provider to also manage an
+ * actively-scheduled OpenTracing Span.
+ * <p>
+ * In any execution context (or any thread, etc), there is at most one "active" Span primarily responsible for the
+ * work accomplished by the surrounding application code. That active Span may be accessed via the
+ * {@link Scheduler#active()} method. If the application needs to defer work that should be part of the same
+ * Span, the Scheduler provides a {@link Scheduler#captureActive()} method that returns a
+ * {@link Continuation}; this activation context may be used to re-activate and deactivate the
+ * captured Span in that other asynchronous executor and/or thread.
+ */
+public interface Scheduler {
+
+    /**
+     * A Continuation can be used *once* to activate a Span and other execution context, then deactivate once the
+     * active period has concluded. (In practice, this active period typically extends for the length of a deferred
+     * async closure invocation.)
+     * <p>
+     * Most users do not directly interact with Continuation, activate(), or deactivate(), but rather use
+     * Scheduler-aware Runnables/Callables/Executors. Those higher-level primitives need not be defined within the
+     * OpenTracing core API.
+     *
+     * @see Scheduler#capture(Span)
+     */
+    interface Continuation extends AutoCloseable {
```

Comment:
This is designed to make it efficient to capture things like MDC state (cc @mabn)